atom-feed/src/main/java/org/openmrs/module/appointments/advice/AtomFeedAdviceProvider.java

```diff
@@ -0,0 +1,33 @@
+package org.openmrs.module.appointments.advice;
+
+import org.aopalliance.aop.Advice;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.appointments.service.AppointmentRecurringPatternService;
+import org.openmrs.module.appointments.service.AppointmentServiceDefinitionService;
+import org.openmrs.module.appointments.service.AppointmentsService;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+/**
+ * Registers AOP advice when the context starts up.
+ * This is done instead of declaring advice in config.xml in order to allow it to be conditionally loaded
+ */
+public class AtomFeedAdviceProvider implements ApplicationContextAware {
```

Comment:
How is this wired up? through component scan? 
Also should the advices be wired up conditionally, since the intention is to work without the atomfeed module? should @Conditional annotation be used?

Reply:
Is there a preferred approach to implementing such conditional loading? I was hoping this had been resolved so that i learn from it